src/services/SendEmailProcessor.ts

```diff
@@ -44,15 +49,62 @@ export class SendEmailProcessor {
 
 	async processRequest(eventBody: any): Promise<EmailResponse> {
 		const email = Email.parseRequest(JSON.stringify(eventBody.Message));
+		// Validate Email model
+		try {
+			await this.validationHelper.validateModel(email, this.logger);
+		} catch (error) {
+			this.logger.error("Failed to Validate Email model data", { messageCode: MessageCodes.MISSING_MANDATORY_FIELDS });
+			throw new AppError(HttpCodesEnum.SERVER_ERROR, "Failed to Validate Email model data.");
+		}
 
-		await this.validationHelper.validateModel(email, this.logger);
+		//Retrieve session event record for the userId
+		let session;
+		try {
+			session = await this.iprService.getSessionBySub(email.userId);
+			this.logger.debug("Session retrieved from session store");
```

Comment:
Shall we `logger.appendKeys({ sessionId: session.session_id });` if we have it to help debugging?

Reply:
No we cant cause in IPV userId is unique identifier to logged in user and it qualifies as PII data.

Reply:
Hmm ok yeah when we had a similar case in CIC Darren confirmed that it was fine to log https://github.com/alphagov/di-ipv-cri-cic-api/pull/236/commits/c478781d35359475048f350aece0181850f4fbf4#r1242358056 but I don't know this service as well as you so happy to take your steer